viahtml/stats.py

```diff
@@ -0,0 +1,108 @@
+"""uWSGI -> New Relic metrics collection logic."""
+
+from collections import defaultdict
+from logging import getLogger
+
+import requests
+from requests import RequestException
+
+LOG = getLogger(__name__)
+
+
+# pylint: disable=missing-yield-doc,missing-yield-type-doc
+
+
+class UWSGINewRelicStatsGenerator:
+    """A callable which returns stats from uWSGI stats for New Relic."""
+
+    # Mappings from various parts of the JSON structure we get back from
+    # uWSGI stats end-point to some suitable metrics names for New Relic
+    ROOT_STATS = {
+        "load": "System/Load",
+        "listen_queue": "ListenQueue/Size",
+        "listen_queue_errors": "ListenQueue/Errors",
+        "signal_queue": "SignalQueue/Size",
+    }
```

Comment:
These mappings are all total guess work, the meanings of the values are totally undocumented